sdk/core/Azure.Core/src/Pipeline/HttpClientTransport.Request.cs

```diff
@@ -18,29 +18,65 @@ public partial class HttpClientTransport : HttpPipelineTransport, IDisposable
     {
         private sealed class HttpClientTransportRequest : Request
         {
+            private readonly PipelineRequest _pipelineRequest;
```

Comment:
As sketched in https://gist.github.com/annelo-msft/642d85ed35b151597197061355c02f06, the only way to get access to the implementation of HttpPipelineRequest in ClientModel from HttpClientTransportRequest in Azure.Core is by adapting the PipelineRequest, even though Azure.Core.Request inherits from PipelineRequest itself.